feat: unique key pretty serialisation

grumbach opened the pull request:
## Description

BREAKING CHANGE: changes serialisation format for unique_key

reviewpad:summary

davidrusu commented on sn_node/src/node/flow_ctrl/cmd_ctrl.rs:
```diff
@@ -280,7 +255,7 @@ impl std::hash::Hash for EnqueuedJob {
         //self.job.hash(state);
         self.job.id().hash(state);
-        self.retries.hash(state);
+        // self.retries.hash(state);
```
```suggestion
```

davidrusu commented on sn_node/src/node/flow_ctrl/cmd_ctrl.rs:
```diff
@@ -264,7 +239,7 @@ impl MsgThroughput {
 pub(crate) struct EnqueuedJob {
     job: CmdJob,
-    retries: usize,
+    // retries: usize,
```
```suggestion
```

davidrusu commented on sn_node/src/node/flow_ctrl/mod.rs:
```diff
@@ -47,30 +50,152 @@ const DYSFUNCTION_CHECK_INTERVAL: Duration = Duration::from_secs(5);
+
+        // the internal process loop
+        loop {
+            debug!("Internal looop");
```
Do we sleep ever? this looks like it will busy loop

davidrusu commented:
Ah missed the `DO NOT MERGE` tag, apologies for the early review

joshuef commented:
Closed as this was incorporated into the original PR #1402

iancoleman committed:
feat(cli): get-faucet has maid address option

This introduces a second way to get tokens from the faucet using the cli

`safe wallet get-faucet <url>`
Retains the existing fixed amount of tokens into the current wallet.

`safe wallet get-faucet <url> <maid_address> <signature>
Receives a transfer from the maid snapshot for the amount of maid owned
by that address. The signature is a bitcoin signature by the maid
address with the message as the hex encoded safe wallet address that
will receive the transfer.

maqi committed:
feat(node): remove bad_node from self RT on detection

joshuef committed:
ci: add cargo-publish dry-run step

This should ensure we don't miss errors that could affect publishing

joshuef committed:
feat!: rename sn-node-manager crate to sn_node_manager

maqi committed:
feat(node): exclude far bad_nodes from RT

JasonPaulGithub committed:
chore: unit test for calculate cost for records

joshuef committed:
chore: remove duplicate token supplies from release-plz toml

joshuef committed:
chore: make sn_node_manager named sn-node-manger once more because crates.io

joshuef committed:
chore(release): sn_cli-v0.89.82-alpha.0/sn_networking-v0.13.31-alpha.0/sn_transfers-v0.16.3-alpha.0/sn_faucet-v0.3.82-alpha.0/sn_metrics-v0.1.3-alpha.0/sn_node-v0.104.37-alpha.0/sn-node-manager-v0.5.0-alpha.0/sn_node_rpc_client-v0.4.69-alpha.0/token_supplies-v0.1.44-alpha.0/sn_client-v0.104.29-alpha.0/sn_protocol-v0.15.2-alpha.0

joshuef committed:
ci: remove the cargo publish check

It relies on prior published versions and _not_ the local versions which means APIs can be mismatched

grumbach committed:
feat: pretty serialisation for unique keys

BREAKING CHANGE: changes serialisation format for unique keys
